addons/purchase/models/purchase.py

```diff
@@ -162,6 +150,13 @@ def create(self, vals):
             vals['name'] = self.env['ir.sequence'].next_by_code('purchase.order') or '/'
         return super(PurchaseOrder, self).create(vals)
 
+    @api.multi
+    def write(self, vals):
+        for order in self:
+            if vals.get('date_planned'):
+                self.mapped('order_line').write({'date_planned': vals.get('date_planned')})
+        return super(PurchaseOrder, self).write(vals)
```

Comment:
don't add the api multi
could batch more
``` python
    def write(self, vals):
        if vals.get('date_planned'):
            self.mapped('order_line').write({'date_planned': vals.get('date_planned')})
        return super(PurchaseOrder, self).write(vals)
```

don't you need to check hte sate of the po line before writing the date_planned?